src/main/java/org/manifold/compiler/front/ExpressionGraph.java

```diff
@@ -52,17 +49,18 @@ public void addVertex(VariableIdentifier vID)
   }
   public void addVertex(ExpressionVertex v) {
     nonVariableVertices.add(v);
+    allVertices.add(v);
   }
 
-  public List<ExpressionVertex> getVertices() {
-    List<ExpressionVertex> vertices = new LinkedList<>();
-    vertices.addAll(getVariableVertices().values());
-    vertices.addAll(getNonVariableVertices());
-    return vertices;
+  public Collection<ExpressionVertex> getVertices() {
+    return allVertices;
   }
 
   private List<ExpressionEdge> edges = new ArrayList<>();
   public void addEdge(ExpressionEdge e) {
+    Preconditions.checkArgument(getVertices().contains(e.getSource())
+                    && e.getTarget() == null || getVertices().contains(e.getTarget()),
+            "Edge had unexpected vertices " + e.toString());
```

Comment:
Arguably free, though. Less wasted vertical space = more of a function / functions displayed on screen = don't need to scroll up/down as much. This is getting too nitty, and tbh I don't really care too much for it. New diff incoming, I'll put it on its own line.